### PR TITLE
Update and cleanup JetCalibrator

### PR DESCRIPTION
**Pull request by ntadej**
This updates JetCalibrator to the new recommendations available since 21.2.41. JES and JER tools are now merged.

This MR:
 - removes all JER references
 - calibration areas are now optional
 - MC type for uncertainties is automatically set
 - default config files are updated

### CODE REVIEW

File: Root/JetCalibrator.cxx [hunk 1]
```diff
@@ -163,13 +163,17 @@ EL::StatusCode JetCalibrator :: initialize ()
...
       ANA_MSG_WARNING( "Could not access simulation flavour from EL::Worker. Treating MC as FullSim by default!" );
     } else {
       m_isFullSim = (stringMeta == "AFII") ? false : true;
     }
     if ( !m_isFullSim ) {
       m_calibConfig = m_calibConfigAFII;
+      m_uncertMCType = "AFII";
+    } else {
+      // Insitu should not be applied to the trimmed jets, per Jet/Etmiss recommendation
+      if ( m_forceInsitu && m_calibSequence.find("Smear") == std::string::npos) m_calibSequence += "_Smear";
```
Comment (jbossios) [inline Root/JetCalibrator.cxx]: @ntadej  Can we use a m_forceSmear instead of m_forceInsitu?

**Comment by ntadej** [comment]
Forgot that .50 does not work due to MET interface changes. Will remove it.

**Comment by mswiatlo** [comment]
At this point the jet calibrations are physically meaningless in anything below 51 (see Steven Schramm's recent emails). I would purge compatibility with pre-51, or else wrap around the MET tool in such a way that this can be updated.

**Comment by kratsg** [comment]
> At this point the jet calibrations are physically meaningless in anything below 51 (see Steven Schramm's recent emails). I would purge compatibility with pre-51, or else wrap around the MET tool in such a way that this can be updated.

I guess we should just bump straight to 21.2.51. @ntadej can you do that?

**Comment by kratsg** [comment]
I've updated this to 21.2.51. when tests pass, ill merge in.

File: xAODAnaHelpers/JetCalibrator.h [hunk 2]
```diff
@@ -57,24 +57,22 @@ class JetCalibrator : public xAH::Algorithm
...
-  std::string m_calibSequence = "JetArea_Residual_Origin_EtaJES_GSC";
-  /// @brief CalibArea tag on CVMFS (for pointing at an area outside of the current release)
-  std::string m_calibArea = "00-04-81"; // current as of 19/11/17
-  /// @brief config for JES Uncertainty Tool
-  std::string m_JESUncertConfig = "";
-  /// @brief JetUncertaintiesTool parameter
-  std::string m_JESUncertMCType = "MC16";
-  /// @brief JES CalibArea tag
-  std::string m_JESCalibArea = "CalibArea-02";
+  std::string m_calibConfigAFII = "JJES_MC16Recommendation_AFII_EMTopo_April2018_rel21.config";
```
Comment (lschaefer) [inline xAODAnaHelpers/JetCalibrator.h]: I think the double J is a typo and it should rather be "JES_MC16Recommendation_AFII_EMTopo_April2018_rel21.config"

**Comment by lschaefer** [comment]
Hi all,
I put a comment below an individual line, but I'm not sure it will show up in the overall discussion so I'll add it here too: I think there is a typo in the JES config file name ("JJES..." instead of "JES"). Thanks!

**Pull request merged**